File: frontend/src/types/channel.ts
/*
 * Сгенерировано `pnpm -C frontend typegen` из docs/schema/channel.schema.json (Pydantic-модели бэкенда).
 * Не править руками: меняется модель, затем команда выше.
 * @generated sha256:65b74ae5a9386a0c259bfc212f2b335936079dc823e3ff76b80d51968f0e0e36
 */
/**
 * Лимиты в USD: конфигурация, не накопление (накопление — в микродолларах, L-001).
 */
export interface Budgets {
  monthly_usd: number
  per_episode_usd: number
  animation_usd: number
}

export interface ChannelProfile {
  id: 'cursus' | 'otto'
  name: string
  format: 'every_rank' | 'host'
  budgets: Budgets
  voice_quota: VoiceQuota
}

export interface VoiceQuota {
  limit_chars: number
}

File: backend/app/models/channel.py
"""Профиль канала — `data/channels/<channel>/profile.json`.

Источник правды по каналу: имя, формат, бюджеты (принцип 6), квота голоса. Таблица `channels`
в SQLite — только реестр для внешних ключей. Потраченное и использованное — состояние, не
профиль: считается из `cost_ledger` и ответов провайдера (M2.7).
"""

from pydantic import Field

from app.models.director import Channel, EpisodeFormat, StrictModel


class Budgets(StrictModel):
    """Лимиты в USD: конфигурация, не накопление (накопление — в микродолларах, L-001)."""

    monthly_usd: float = Field(gt=0)
    per_episode_usd: float = Field(gt=0)
    animation_usd: float = Field(ge=0)


class VoiceQuota(StrictModel):
    limit_chars: int = Field(gt=0)


class ChannelProfile(StrictModel):
    id: Channel
    name: str
    format: EpisodeFormat
    budgets: Budgets
    voice_quota: VoiceQuota
